File: backend/src/main/java/edu/cit/spot/entity/Teacher.java
package edu.cit.spot.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@Table(name = "teachers")
public class Teacher {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank
    @Size(max = 50)
    private String firstName;
    
    @Size(max = 50)
    private String middleName;
    
    @NotBlank
    @Size(max = 50)
    private String lastName;
    
    @NotBlank
    @Email
    @Size(max = 100)
    @Column(unique = true)
    private String email;
    
    @NotBlank
    @Size(max = 50)
    private String teacherPhysicalId;
    
    @NotBlank
    @Size(max = 100)
    private String password;
    
    private String googleId;
    
    private boolean googleLinked = false;
    
    @OneToMany(mappedBy = "teacher", fetch = FetchType.LAZY)
    private List<Section> sections = new ArrayList<>();
}

File: frontend_mobile/app/src/main/java/com/example/spot/model/Teacher.kt
package com.example.spot.model

/**
 * Teacher data model matching the backend TeacherDto
 */
data class Teacher(
    val id: Long,
    val firstName: String,
    val middleName: String?,
    val lastName: String,
    val email: String,
    val teacherPhysicalId: String,
    val googleLinked: Boolean,
    val assignedSectionIds: List<Long>
) {
    // Computed property for full name
    val name: String
        get() = if (middleName.isNullOrEmpty()) {
            "$firstName $lastName"
        } else {
            "$firstName $middleName $lastName"
        }
}
